### ISSUE TITLE
[tf.data] _pywrap_server_lib.so breaks nightly packages

### ISSUE THREAD

**Issue by byronyi** [tensorflow/tensorflow#38153]
Our in-house nightly builds were broken since 2020-03-31 when auditwheel tries to repair the nightly packages. The reason under the hood seems to be an incorrect link from the recent change of adding tf.data service support in 0b63458060d7f8400c7e5b37ccdd7ca89c2ffbe1 and enabled in nightly packages in 503179f662708e730fd0a0c98261e9678975e3b8.

Install the latest nightly, and check the dynamic linker:

```
$ ldd /usr/local/lib/python3.8/dist-packages/tensorflow/core/data/service/python/_pywrap_server_lib.so
	linux-vdso.so.1 (0x00007ffd621f4000)
	libtensorflow_framework.so.2 => /usr/local/lib/python3.8/dist-packages/tensorflow/core/data/service/python/../../../../libtensorflow_framework.so.2 (0x00007fda32b70000)
	_pywrap_tensorflow_internal.so => not found
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fda32b5a000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fda32b39000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fda329f2000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fda32825000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fda3280b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fda32648000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fda34bfd000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fda3263d000)
```

Obviously it links to `_pywrap_tensorflow_internal.so` but it is not found with the relative path.

PS: we are using `auditwheel==3.0.0` to produce manylinux2014 builds, but the official tf-nightly uses an older version which fails to catch this.

Ping @aaudiber @mihaimaruseac. We had met a similar issue in #36067. As mentioned there, the reason seems to be that ``tf_python_pybind_extension`` should only be used for packages in ``//tensorflow/python/...`` but it is used in ``//tensorflow/core/..``  in this case.

Any idea how to avoid this in the future? Also cc @av8ramit.

**Comment by aaudiber** [comment]
Thank you @byronyi for investigating this. I just put out a CL to move the usage of `tf_python_pybind_extension` under `//tensorflow/python`. I will validate the fix and merge it as soon as possible.

**Comment by byronyi** [comment]
@aaudiber Thanks for the prompt fix! Can't wait to try out tf.data service :)

**Comment by byronyi** [comment]
This should be fixed by 8434a67458fe52ffb9f978960bc81e2396ad76d3. Thanks!
